lib/fluent/winsvc.rb

```diff
@@ -57,8 +59,16 @@ def initialize(service_name)
     
     def service_main
       @pid = service_main_start(@service_name)
-      while running?
-        sleep 10
+      begin
+        loop do
+          sleep 5
+          break unless running?
+          pid, status = Process.waitpid2(@pid, Process::WNOHANG)
+          raise Errno::ECHILD if pid || status
+        end
+      rescue Errno::ECHILD
+        set_event(@@hStopEvent)
```

Comment:
```suggestion
        SetEvent(@@hStopEvent)
```

We should use [SetEvent of Win32 api](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-setevent) here.

`set_event` is a private method of this class to signal to the supervisor.

Reply:
Although I don't think it's supervisor specific, using `set_event` here isn't suitable as you say.
`set_event` expects a event name as the argument but `@@hStopEvent` is an event handle itself.
I'll fix it.

Reply:
> Although I don't think it's supervisor specific, 

Oh! Sorry, that's right!

Reply:
Done